bin/faucet/src/main.rs

```diff
@@ -501,11 +486,11 @@ mod test {
 
         // Wait for faucet to be up
         let endpoint = Url::parse("http://localhost:8080").unwrap();
-        let addr = endpoint.to_socket().unwrap();
+        let addr = endpoint.socket_addrs(|| None).unwrap();
         let start = Instant::now();
         let timeout = Duration::from_secs(10);
         loop {
-            match tokio::net::TcpStream::connect(addr).await {
+            match tokio::net::TcpStream::connect(&*addr).await {
```

Comment:
nit: I think you can do `&addrs[..]` here instead of `&*addr`. I'd prefer the first one because it's a bit more clear than the latter, that relies a bit on auto dereferencing